### PR TITLE
chore(bench): add codspeed bench

### PR DESCRIPTION
**Pull request by adriencaccia**
Add [CodSpeed](https://codspeed.io/) continuous benchmarking tool.

In order to make it work, the `serverless-contracts` package has been switched to `esm`. This is something that needed to be done at some time, so it is a nice opportunity to start experimenting with it 😄

### CODE REVIEW

**Comment by adriencaccia** [comment]
@fargito WDYT?

Especially the part about `esm`. I tested it on my end and found it's working. I would like your opinion/tests before merging this 😉

File: packages/serverless-contracts/package.json
Comment (fargito) [inline packages/serverless-contracts/package.json:37]: Why was it necessary to remove the `--validate` option?

File: packages/serverless-contracts/package.json
Comment (fargito) [inline packages/serverless-contracts/package.json:27]: Maybe this folder should be outside of `src`?
We need to make sure this is not included in the code shipped to npm

**Review by fargito — CHANGES_REQUESTED** [review]
Very nice!
Let's ensure we don't ship the benches to npm though

File: packages/serverless-contracts/package.json
Comment (adriencaccia) [reply packages/serverless-contracts/package.json:37]: It is explained here: https://github.com/sverweij/dependency-cruiser/blob/develop/doc/cli.md#--config---validate

Basically, `--validate` and `--config` are aliases, and we need to use it to have the default config (in our case `.dependency-cruiser.cjs`) loaded

File: packages/serverless-contracts/package.json
Comment (adriencaccia) [reply packages/serverless-contracts/package.json:27]: Since nothing from this folder is exported from `src/index.ts`, there will not be any `js` code exported. And for types the `src/__benches__` folder is excluded from the `tsconfig.build.json`.

I wanted to keep the benches in the `src` folder because, in the long run, our benches will be done through `vitest bench` (once CodSpeed supports it), and our benches will live right next to the source code as tests do.

WDYT?

File: packages/serverless-contracts/package.json
Comment (fargito) [reply packages/serverless-contracts/package.json:37]: Perfect!

File: packages/serverless-contracts/package.json
Comment (fargito) [reply packages/serverless-contracts/package.json:27]: I see, and in this case `src/__benches__/index.ts` will disappear right ? In order to minimize the rework when that is done, we can put it in `__benches__` and directly import from all the benches subdirectories (skip all the `index.ts` exports). WDYT?

Comment (adriencaccia) [reply packages/serverless-contracts/package.json:27]: Yes you are completely right, I will do that 😉

**Review by fargito — APPROVED** [review]

**Pull request merged**
